File: 프로그래머스/lv0/120833. 배열 자르기/배열 자르기.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>

int* solution(int numbers[], size_t numbers_len, int num1, int num2) {
    int len = num2 - num1 + 1;
    
    int* answer = (int*)malloc(len * sizeof(int));
    if (answer == NULL) return NULL;
    
    for (int i = num1, j = 0; i <= num2; i++, j++) {
        answer[j] = numbers[i];
    }
    
    return answer;
}

File: 프로그래머스/0/120833. 배열 자르기/배열 자르기.java
class Solution {
    public int[] solution(int[] numbers, int num1, int num2) {
        int len = num2 - num1 + 1;
        
        int[] answer = new int[len];
        
        for (int i = 0, j = num1; j <= num2; j++) {
            answer[i++] = numbers[j];
        }
        
        return answer;
    }
}